parser.go

```diff
@@ -552,10 +555,16 @@ func (p *parser) parseType() (Type, error) {
 			if err != nil {
 				return nil, err
 			}
-			scale, err := strconv.Atoi(sScale)
-			if err != nil {
-				// unexpected since sIndex should conform to pIndex
-				panic(err)
+			var scale int = 32 + 1
+			if len(sScale) <= len("32") {
+				scale, err = strconv.Atoi(sScale)
+				if err != nil {
+					// unexpected since sIndex should conform to pIndex
+					panic(err)
+				}
+			}
+			if scale > 32 {
```

Comment:
I'd be +1 on farming this (`32`) out to a constant somewhere, so it's easy to reference consistently.

Reply:
Yes, we definitely should. We also need to apply this consistently so that `round down 33` errors out with the same message around scale being to large. I'll address as part of this change, might as well.